index.js

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env node
+
+const axios = require('axios');
+const path = require('path');
+const fs = require('fs-extra');
+const program = require('commander');
+
+let input;
+
+program
+    .name('icaredata-client')
+    .usage('<path-to-messages> [options]')
+    .option('-o, --out <out>', 'the path to the output folder', path.join('.', 'out'))
+    .arguments('<path-to-messages>')
+    .action(function (pathToMessages) {
+        input = pathToMessages;
+    })
+    .parse(process.argv);
+
+// Check that input folder is specified
+if (!input) {
+    console.error('Missing path to messages folder.');
+    program.help();
+}
+
+// Try to parse the files from the input directory
+let files;
+try {
+    files = fs.readdirSync(input, 'utf8');
+} catch (e) {
+    console.error('Invalid path to messages folder.');
+    program.help();
+}
+
+// Collect the message files and the config file
+const data = { config: null, messages: [] };
+for (const file of files) {
+    const fileContent = fs.readFileSync(path.join(input, file), 'utf8');
+    if (fileContent) {
+        if (file === 'config.json') {
+            data.config = JSON.parse(fileContent);
+        } else if (file.endsWith('.json')) {
+            data.messages.push(fileContent);
+        }
+    }
+}
+
+if (!data.config) {
+    console.error('No config.json in messages folder.');
+    program.help();
+}
+
+if (data.messages.length === 0) {
+    console.error('No messages found in messages folder.');
+    program.help();
+}
+
+const apiGateway = axios.create({
+    baseURL: data.config.baseURL,
+    timeout: data.config.timeout,
+});
+
+Promise.all(data.messages.map(message => {
+    apiGateway.post('/DSTU2/$process-message', message)
+    .then(r => console.log('Success'))
+    .catch(e => console.error(e.message));
+}));
```

Comment:
I'm not positive what's happening here, maybe I'm just unaware of some functionality that `Promise.all` has, but I don't think you need `Promise.all`. `Promise.all` waits for all of the promises it receives to resolve before going to the `then` clause and (I believe) `catch`es if any of the promises reject. However, within your `map` callback, you aren't actually returning anything, so the array that you're passing into `Promise.all` doesn't contain any promises. In that case I'm not really sure what happens, but it seems like nothing. Is this intentional? Also, 
```javascript
apiGateway.post('/DSTU2/$process-message', message)
    .then(r => console.log('Success'))
    .catch(e => console.error(e.message));
```
isn't a promise either because you're `then`ing it without returning a promise in the `then`. So even if you were to return in `map`'s callback, you still wouldn't get any promises. The promise clause is `apiGateway.post('/DSTU2/$process-message', message)`, so you'd have to return that and handle resolutions and rejections from the `Promise.all` promise. At least that's how I've always used `Promise.all`.

So from my understanding, I think if you actually do want Promise.all, you would do something like
```javascript
Promise.all(data.message.map(message => apiGateway.post('/DSTU2/$process-message', message)))
.then(responses => {
    // do something
}).catch(error => {
    // do something
})
```
Otherwise, I think you just want
```javascript
for (const message of data.messages) {
    apiGateway.post('/DSTU2/$process-message', message)
        .then(r => console.log('Success'))
        .catch(e => console.error(e.message));
}
```
Or I could be completely misunderstanding what's happening, sorry if I am!

Reply:
Thanks for the explanation. You aren't misunderstanding, I think I was just trying to follow a template that I did on an old project without fully understanding!

Yeah, I actually initially just had the loop that you suggested at the end and it worked fine. I'll revert since I just overcomplicated it for no reason.